### DISCUSSION TITLE
Feature Request: Day-of-Month Parameter for Timers

### DISCUSSION THREAD

**Discussion by ZhuDaHai** [arendst/Tasmota#17800] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
This is a feature Request: I would like it if a Day-Of-Month option was added to the Timers function so that, as an alternate to Weekday selections, a day (or days) of the month could be entered for when to execute the Timer.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
  I have opened a discussion on the Discussions forum (#17792) and received wonderful responses, but short of compiling a new binary with added features (scripting,rule logic, cron, etc), there doesn't seem to be a straightforward way to accomplish this.
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Sonoff Basic
- [x] Tasmota binary firmware version number used: 12.3.1
  - [x] Pre-compiled
  - [ ] Self-compiled
- [x] Flashing tools used: Upgrade by File Upload
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
09:55:10.677 CMD: Backlog Template; Module; GPIO 255
09:55:10.742 MQT: stat/sonoff-01/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,4704],"FLAG":0,"BASE":18}
09:55:10.951 MQT: stat/sonoff-01/RESULT = {"Module":{"1":"Sonoff Basic"}}
09:55:11.158 MQT: stat/sonoff-01/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}

```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
09:55:52.244 CMD: Backlog Rule1; Rule2; Rule3
09:55:52.327 MQT: stat/sonoff-01/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":70,"Free":441,"Rules":"on Tele-Time#Data$>:50 DO backlog power1 0; delay 200; power1 1; endon"}}
09:55:52.538 MQT: stat/sonoff-01/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
09:55:52.739 MQT: stat/sonoff-01/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
09:56:53.567 CMD: Status 0
09:56:53.574 MQT: stat/sonoff-01/STATUS = {"Status":{"Module":1,"DeviceName":"TaskLamp","FriendlyName":["TaskLamp"],"Topic":"sonoff-01","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":1,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
09:56:53.580 MQT: stat/sonoff-01/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"sonoffs","OtaUrl":"http://thehackbox.org/tasmota/release/sonoff.bin","RestartReason":"Software/System restart","Uptime":"0T13:58:56","StartupUTC":"2023-01-26T11:57:57","Sleep":50,"CfgHolder":4617,"BootCount":153,"BCResetTime":"2023-01-26T09:32:35","SaveCount":4177,"SaveAddress":"F6000"}}
09:56:53.585 MQT: stat/sonoff-01/STATUS2 = {"StatusFWR":{"Version":"12.3.1(tasmota)","BuildDateTime":"2022-12-16T10:28:17","Boot":7,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"388/699"}}
09:56:53.589 MQT: stat/sonoff-01/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Loki",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008029","2805C80001000600003C5AFF002800000000","00000080","00006000","00004000","00000000"]}}
09:56:53.602 MQT: stat/sonoff-01/STATUS4 = {"StatusMEM":{"ProgramSize":629,"Free":372,"Heap":22,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
09:56:53.609 MQT: stat/sonoff-01/STATUS5 = {"StatusNET":{"Hostname":"Sonoff-1","IPAddress":"172.27.3.11","Gateway":"172.27.3.3","Subnetmask":"255.255.255.0","DNSServer1":"172.27.3.3","DNSServer2":"0.0.0.0","Mac":"DC:4F:22:92:42:71","Webserver":2,"HTTP_API":1,"WifiConfig":2,"WifiPower":17.0}}
09:56:53.615 MQT: stat/sonoff-01/STATUS6 = {"StatusMQT":{"MqttHost":"172.27.3.4","MqttPort":1883,"MqttClientMask":"work_lamp","MqttClient":"work_lamp","MqttUser":"MQTT_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
09:56:53.622 MQT: stat/sonoff-01/STATUS7 = {"StatusTIM":{"UTC":"2023-01-27T01:56:53","Local":"2023-01-27T09:56:53","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+08:00","Sunrise":"06:48","Sunset":"17:23"}}
09:56:53.626 MQT: stat/sonoff-01/STATUS10 = {"StatusSNS":{"Time":"2023-01-27T09:56:53"}}
09:56:53.632 MQT: stat/sonoff-01/STATUS11 = {"StatusSTS":{"Time":"2023-01-27T09:56:53","Uptime":"0T13:58:56","UptimeSec":50336,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"Loki","BSSId":"30:23:03:B7:0D:FA","Channel":1,"Mode":"11n","RSSI":84,"Signal":-58,"LinkCount":1,"Downtime":"0T00:00:03"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
No issues, feature request

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
It does not appear to be possible to set a specific day (or days) of the month for a timer to execute. Only day(s) of the week.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
I wish to add a day of month (or days of month) option to the Timer function so that instead of setting a day-of-week (or days-of-the week) to execute the timer, I can select which day(s) of the month to execute it - all with the same options of the standard timers.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
This option does not need to be available in the Timers UI, but could be added an an option to the Timer[x] commands in the Console

I realize this affect can be done with a custom built binary with any number of features enabled (scripting, rule logic, cron jobs, etc.), but I think it has enough merrit to be added as a standard feature (IMO, maybe I'm the only one who wants this). And I'm not very savvy at compiling this code at this point.

**(Please, remember to close the issue when the problem has been addressed)**
Thank you for your time and consideration.

**Comment by cruiser2** [comment]
I support this request.
